src/main/scala/is/hail/variant/VariantSampleMatrix.scala

```diff
@@ -1236,6 +1236,12 @@ class VariantSampleMatrix[RPK, RK, T >: Null](val hc: HailContext, val metadata:
 
   def hadoopConf: hadoop.conf.Configuration = hc.hadoopConf
 
+  def head(n: Long = 10): VariantSampleMatrix[RPK, RK, T] = {
+    if (n < 0)
+      fatal(s"n must be positive! Found `$n'.")
+    copy(rdd = rdd.head(n).toOrderedRDD)
```

Comment:
don't call toOrderedRDD - you should implement `head` on OrderedRDD, which will call RichRDD.head and then subset the range bounds as needed.